src/wpool_process_callbacks.erl

```diff
@@ -0,0 +1,76 @@
+-module(wpool_process_callbacks).
+
+-behaviour(gen_event).
+
+%% gen_event callbacks
+
+-export([ init/1
+        , handle_event/2
+        , handle_call/2
+        , handle_info/2
+        , code_change/3
+        , terminate/2]).
+
+-export([notify/3]).
+-type state() :: ordset:ordset(module()).
+
+-type event() :: on_init_start | on_new_worker | on_worker_dead.
+
+-spec init([wpool:option()]) -> {ok, state()}.
+init(WPoolOpts) ->
+{ok, maybe_initial_callbacks(WPoolOpts)}.
+
+-spec handle_event({event(), [any()]}, state()) -> {ok, state()}.
+handle_event({Event, Args}, Callbacks) ->
+  [call(Callback, Event, Args) || Callback <- Callbacks],
+  {ok, Callbacks};
+handle_event(_, State) ->
+  {ok, State}.
+
+-spec handle_call(Call, state()) -> {ok, ok, state()} when
+    Call :: {add_callback, module()} | {remove_callback, module()}.
+handle_call({add_callback, Module}, Callbacks) ->
+  {ok, ok, ordsets:add_element(Module, Callbacks)};
+handle_call({remove_callback, Module}, Callbacks) ->
+  {ok, ok, ordsets:del_element(Module, Callbacks)};
+handle_call(_, State) ->
+  {ok, ok, State}.
+
+handle_info(_, State) ->
+  {ok, State}.
+
+code_change(_OldVsn, State, _Extra) ->
+  {ok, State}.
+
+terminate(_Reason, _State) ->
+  ok.
+
+-spec notify(event(), [wpool:option()], [any()]) -> ok.
+notify(Event, Options, Args) ->
+  case lists:keyfind(event_manager, 1, Options) of
+    {event_manager, EventMgr} ->
+      gen_event:notify(EventMgr, {Event, Args});
+    _ ->
```

Comment:
Considering we're forcibly adding this element to the worker options for all pools, I don't see how this can return anything other than `{event_manager, EventMgr}`.

Reply:
I wanted (and already made) the event manager optional. It get's started only if `callbacks` are passed when starting the pool or (if we want to add callbacks later) if the `enabled_callbacks` option is set to true.